Clarified auth docs (#13000)
directus/directus: docs/configuration/config-options.md

| ---------------------- | ----------------------------------------- | ------------- |
| `AUTH_PROVIDERS`       | A comma-separated list of auth providers. | --            |
| `AUTH_DISABLE_DEFAULT` | Disable the default auth provider         | `false`       |

For each auth provider you list, you must also provide the following configuration:

| Variable                 | Description                                                     | Default Value |
| ------------------------ | --------------------------------------------------------------- | ------------- |
| `AUTH_<PROVIDER>_DRIVER` | Which driver to use, either `local`, `oauth2`, `openid`, `ldap` | --            |

You may also be required to specify additional variables depending on the auth driver. See configuration details below.

::: warning Multiple Providers

Directus users can only authenticate using the auth provider they are created with. It is not possible to authenticate
with multiple providers for the same user.

:::

### Local (`local`)

The default Directus email/password authentication flow.

No additional configuration required.

> ### SSO (`openid` and `oauth2`)

Author: ```suggestion
### SSO (`oauth2` and `openid`)
```

AUTH_GITHUB_PROFILE_URL="https://api.github.com/user"
```

> More examples of SSO configurations [can be found here](/configuration/sso-examples/).

Author: ```suggestion
More example SSO configurations [can be found here](/configuration/sso-examples/).
```

::: warning PUBLIC_URL

These flows rely on the `PUBLIC_URL` variable for redirecting. Ensure the variable is correctly configured.

:::

### OAuth 2.0

| Variable                                    | Description                                                                                    | Default Value    |
| ------------------------------------------- | ---------------------------------------------------------------------------------------------- | ---------------- |

> | `AUTH_<PROVIDER>_CLIENT_ID`                 | Client identifier for the external service.                                                    | --               |
> | `AUTH_<PROVIDER>_CLIENT_SECRET`             | Client secret for the external service.                                                        | --               |
> | `AUTH_<PROVIDER>_SCOPE`                     | A white-space separated list of permissions to request.                                        | `email`          |
> | `AUTH_<PROVIDER>_AUTHORIZE_URL`             | Authorization page URL of the external service.                                                | --               |
> | `AUTH_<PROVIDER>_ACCESS_URL`                | Access token URL of the external service.                                                      | --               |
> | `AUTH_<PROVIDER>_PROFILE_URL`               | User profile URL of the external service.                                                      | --               |

Reviewer: ```suggestion
| `AUTH_<PROVIDER>_CLIENT_ID`                 | Client identifier for the OAuth provider.                                                    | --               |
| `AUTH_<PROVIDER>_CLIENT_SECRET`             | Client secret for the OAuth provider.                                                        | --               |
| `AUTH_<PROVIDER>_SCOPE`                     | A white-space separated list of permissions to request.                                        | `email`          |
| `AUTH_<PROVIDER>_AUTHORIZE_URL`             | Authorization page URL of the OAuth provider.                                                | --               |
| `AUTH_<PROVIDER>_ACCESS_URL`                | Access token URL of the OAuth provider.                                                      | --               |
| `AUTH_<PROVIDER>_PROFILE_URL`               | User profile URL of the OAuth provider.                                                      | --               |
```

> | `AUTH_<PROVIDER>_CLIENT_ID`                 | Client identifier for the external service.                                                    | --               |
> | `AUTH_<PROVIDER>_CLIENT_SECRET`             | Client secret for the external service.                                                        | --               |
> | `AUTH_<PROVIDER>_SCOPE`                     | A white-space separated list of permissions to request.                                        | `email`          |
> | `AUTH_<PROVIDER>_AUTHORIZE_URL`             | Authorization page URL of the external service.                                                | --               |
> | `AUTH_<PROVIDER>_ACCESS_URL`                | Access token URL of the external service.                                                      | --               |
> | `AUTH_<PROVIDER>_PROFILE_URL`               | User profile URL of the external service.                                                      | --               |

Reviewer: ```suggestion
| `AUTH_<PROVIDER>_CLIENT_ID`                 | Client identifier for the OAuth provider.                                                    | --               |
| `AUTH_<PROVIDER>_CLIENT_SECRET`             | Client secret for the OAuth provider.                                                        | --               |
| `AUTH_<PROVIDER>_SCOPE`                     | A white-space separated list of permissions to request.                                        | `email`          |
| `AUTH_<PROVIDER>_AUTHORIZE_URL`             | Authorization page URL of the OAuth provider.                                                | --               |
| `AUTH_<PROVIDER>_ACCESS_URL`                | Access token URL of the OAuth provider.                                                      | --               |
| `AUTH_<PROVIDER>_PROFILE_URL`               | User profile URL of the OAuth provider.                                                      | --               |
```

::: warning PUBLIC_URL

> These flows rely on the `PUBLIC_URL` variable for redirecting. Ensure that variable is configured correctly.

Author: ```suggestion
These flows rely on the `PUBLIC_URL` variable for redirecting. Ensure the variable is configured correctly.
```

:::

### OAuth 2.0

| Variable                                    | Description                                                                                    | Default Value    |
| ------------------------------------------- | ---------------------------------------------------------------------------------------------- | ---------------- |
| `AUTH_<PROVIDER>_CLIENT_ID`                 | Client identifier for the external service.                                                    | --               |
| `AUTH_<PROVIDER>_CLIENT_SECRET`             | Client secret for the external service.                                                        | --               |
| `AUTH_<PROVIDER>_SCOPE`                     | A white-space separated list of permissions to request.                                        | `email`          |
| `AUTH_<PROVIDER>_AUTHORIZE_URL`             | Authorization page URL of the external service.                                                | --               |

> | `AUTH_<PROVIDER>_ACCESS_URL`                | Access token URL of the external service.                                                      | --               |

Author: Should we say "OAuth provider" instead of "external service"?

Reviewer: Yeah lets do it

Reviewer: Yeah lets do it
